rotkehlchen/logging.py

```diff
@@ -90,13 +92,20 @@ def process(self, given_msg: Any, kwargs: MutableMapping[str, Any]) -> tuple[str
         This is the main post-processing function for rotki logs
 
         This function:
-        - appends all kwargs to the final message
+        - appends all kwargs to the final message, redacting any sensitive information
         - appends the greenlet id in the log message
         """
-        msg = str(given_msg)
-        greenlet = gevent.getcurrent()
+        msg, greenlet = str(given_msg), gevent.getcurrent()
         greenlet_name = get_greenlet_name(greenlet)
-        msg = greenlet_name + ': ' + msg + ','.join(f' {a[0]}={a[1]}' for a in kwargs.items())
+        if 'json_data' in kwargs and isinstance(kwargs['json_data'], dict):
+            sanitized_data = kwargs['json_data'].copy()
```

Comment:
question: Do you need to make a copy? Would modifying here also modify the actual args being sent? Did you check?

Reply:
> question: Do you need to make a copy? Would modifying here also modify the actual args being sent? Did you check?

yes. it'll modify the dict.